course/models.py

```diff
@@ -954,16 +960,18 @@ def enroll_from_sis(self) -> Tuple[int, int]:
                 # yet logged in to A+, then the user profile does not exist yet.
                 pass
 
-        # Remove SIS-enrolled students who are not anymore in SIS participants,
-        # for example, because they have first enrolled in SIS, but then
-        # unenrolled themselves.
-        students = self.all_students.filter(enrollment__from_sis=True)
-        to_remove = students.exclude(student_id__in=participants)
-        qs = Enrollment.objects.filter(user_profile__in=to_remove, course_instance=self)
-        qs.update(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
-        for e in qs:
-            invalidate_content(Enrollment, e)
-            delcount += 1
+        # Ignore empty participants list caused by a rare SIS API gateway malfunction
+        if participants:
```

Comment:
How about adding an else branch, with a (non-severe) log message about the empty participants list, and in that case not showing the "enrolled X students" message (not forgetting to return (0,0) in that case)? Later in the course it indicates a likely problem at SISU end, but for courses that have not yet started it may be a valid situation.